clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -3075,6 +3075,55 @@ LogicalResult GetRuntimeMemberOp::verify() {
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// GetMethodCalleeOp Definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult GetMethodCalleeOp::verify() {
+  auto methodTy = getMethod().getType();
+  auto objectPtrTy = mlir::cast<mlir::cir::PointerType>(getObject().getType());
+  auto objectTy = objectPtrTy.getPointee();
+
+  if (methodTy.getClsTy() != objectTy) {
+    emitError() << "method class type and object type do not match";
+    return mlir::failure();
+  }
+
+  auto calleePtrTy = mlir::cast<mlir::cir::PointerType>(getCallee().getType());
+  auto calleeTy = mlir::cast<mlir::cir::FuncType>(calleePtrTy.getPointee());
+  auto methodFuncTy = methodTy.getMemberFuncTy();
+
+  // Assume:
+  // - objectTy is !cir.ptr<!T>
+  // - methodFuncTy is !cir.func<!Ret (!Args)>
+  //
+  // We verify at here that:
+  // - calleeTy is !cir.func<!Ret (!cir.ptr<!T>, !Args)>
+  if (methodFuncTy.getReturnType() != calleeTy.getReturnType()) {
+    emitError() << "method return type and callee return type do not match";
+    return mlir::failure();
+  }
+
+  auto calleeArgsTy = calleeTy.getInputs();
+  auto methodFuncArgsTy = methodFuncTy.getInputs();
+
+  if (calleeArgsTy.empty()) {
+    emitError() << "callee parameter list lacks receiver object ptr";
+    return mlir::failure();
+  }
+  if (calleeArgsTy[0] != objectPtrTy) {
+    emitError()
+        << "the first parameter of callee must be the receiver object ptr";
+    return mlir::failure();
+  }
+  if (calleeArgsTy.slice(1) != methodFuncArgsTy) {
+    emitError() << "callee parameters and method parameters do not match";
+    return mlir::failure();
+  }
```

Comment:
These constraints should be mentioned in the documentation.

Reply:
Updated.